synapse/replication/http/devices.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from synapse.replication.http._base import ReplicationEndpoint
+
+logger = logging.getLogger(__name__)
+
+
+class ReplicationUserDevicesResyncRestServlet(ReplicationEndpoint):
+    """Notifies that a user has joined or left the room
```

Comment:
it doesn't seem to be about a notification? It seems to be a request to fetch the user's devices from the remote server?

We should probably also record here (or at least somewhere in the code comments, rather than just the PR where it will get lost) why it is important for this to happen on the master.

Reply:
Woops, failed to delete the docstring when c+p'ed and then was like "yay i've already written a docstring" :man_facepalming: